pallas-primitives/src/alonzo/model.rs

```diff
@@ -1308,7 +702,11 @@ pub struct WitnessSet {
     pub bootstrap_witness: Option<Vec<BootstrapWitness>>,
 
     #[n(3)]
-    pub plutus_script: Option<Vec<PlutusScript>>,
+    pub plutus_v1_script: Option<Vec<PlutusScript<1>>>,
+
+    #[cbor(skip)]
+    #[deprecated(since = "0.31.0", note = "use `plutus_v1_script` instead")]
+    pub plutus_script: Option<Vec<PlutusScript<1>>>,
```

Comment:
Do we need this change?

This is Alonzo specific and the CDDL names this field as `plutus_script`.

I'm worried that having a field skipped by cbor decoding will be a hard issue to debug.

Reply:
Names don't exist in CBOR; it's only annotation in CDDL so we could pick whatever. I chose to rename it to `plutus_v1_script` for consistency with other eras. I can revert, but as the codebase evolves, I rather keep things tidy than carrying over oddities from the past as it only creates more cognitive overload. 

The skipped field here is only added per your request to not break the API and provide clear deprecation warnings. It is perhaps overkill for struct field? Happy to just drop it.

Reply:
Thanks for explaining how CBOR and CDDL works.

Regardless, I once again state that the goal of the primitives crate is to reflect the CDDL as much as possible so that it might, at some point, be autogenerated (or at least partially).

The primitives crate is NOT the place for ergonomic improvements. For nice names and ergonomic types, we can use traverse or a custom wrapper in Amaru.

Reply:
hm I think trying to reflect the CDDL is silly and I think autogenerating is a pipe dream

Reply:
Maybe we can clarify the CDDL upstream? I was hoping to submit a PR, because in the latest CDDL they've removed the helpful comments that describe what each field in the transaction body is anyway lol: 

https://github.com/IntersectMBO/cardano-ledger/blob/master/eras/conway/impl/cddl-files/conway.cddl#L451-L470

Reply:
> hm I think trying to reflect the CDDL is silly and I think autogenerating is a pipe dream. That said, I'm largely indifferent. But I can say from experience, trusting KtorZ with these kinds of choices is always worth it long term.

I respect your opinion @rvcas and I trust @KtorZ expertise / track-record, but I fail to see why we want to introduce a breaking change that would affect 15+ downstream repos for an arbitrary name change that, IMO, is against the goal of the module.

These CDDLs are as close as we can get to a spec in Cardano. Either via codegen or manually, there needs to be a direct, 1:1 mapping between the spec and the decoding logic. Adding an extra layer of arbitrary naming translation in the serialization layer makes the module much harder to maintain.

I've been maintain this codebase for a few years now. I've been woken at night several times now because some [peculiar block](https://github.com/txpipe/pallas/blob/4871342a8deffaceafb5e0f771e1623dfe57e25f/pallas-primitives/src/alonzo/model.rs#L1615) had a CBOR deserialization error. Having the codec primitives as similar to the CDDL made it easier to find the root cause. I'll be more than happy to introduce any breaking change that brings the two closer together.

I'm also in favor of adding an ergonomic API layer on top of the primitives. That's what the `traverse` crate is for, but I won't oppose to someone building a different one if the design decisions are incompatible.

I try to welcome all contributions and I consider everybody in this discussion more knowledgeable than me, but please don't disregard my feedback as silly because it's coming from direct experience maintaining this codebase.

Reply:
No worries, I wasn't disregarding you or your opinion when I said that. Apologies if it came off negatively. If these were voice messages the tone of my voice would have indicated this but apologies anyways for not being more careful in text format.

Reply:
I believe the right move is what @Quantumplation hinted at:

> Maybe we can clarify the CDDL upstream?

As for the motivation behind this change, simply consistency, which in long run pays off towards maintainability and discoverability. Anyway, since this is sparkling too much debate and we're already at 4000 words spilled for a 3 character change in a now-legacy era, I am going to revert that one!